cluster-api/pkg/controller/machine/zz_generated.api.register.go

```diff
@@ -44,25 +44,15 @@ type MachineController struct {
 }
 
 // NewController returns a new MachineController for responding to Machine events
-func NewMachineController(config *rest.Config, si *sharedinformers.SharedInformers) *MachineController {
+func NewMachineController(config *rest.Config, si *sharedinformers.SharedInformers, actuator Actuator) *MachineController {
```

Comment:
Did you consider exposing `MachineController.controller` field instead of passing an `actuator` as a parameter ? In that case fixing failing build would be easier and in the future we could inject more custom code directly to `MachineImplementation`

Reply:
I much prefer an API contract where I get a fully functioning object back from `New`. Having to further initialize the object after constructing just seems wrong. If anything, we could perhaps pass in the controller instead of the actuator.

Exposing the `.controller` field allows for partial initialization. One case where this is bad is when "cluster-api/pkg/controller/zz_generated.api.register.go" is regenerated to add a new controller. If the machine controller is added back there (which I don't know if it does), it would compile fine, but it would run an uninitialized `MachineController` that would nil panic as soon as it tried to access the `.controller` field. I would much rather deal with the confusion of compilation errors for now.